threads_api/src/threads_api.py

```diff
@@ -230,18 +254,28 @@ async def _set_logged_in_state(username, token):
             self.logger.info("Set logged-in state successfully. All set!")
             return
         
-        if username is None or password is None:
-            raise Exception("Username or password are invalid")
+        if not (username := username or self.username):
+            username = input("Please enter your Instagram username: ")
+
+        if not (password := password or self.password):
+            password = getpass("Please enter your Instagram password: ")
+
+        if not (username and password):
+            raise Exception("No username or password provided.")
 
         self.username = username
+        self.password = password
 
+        cached_token_path = cached_token_path or self.cached_token_path
+        self.cached_token_path = cached_token_path
+                
         # Look in cache before logging in.
         if cached_token_path is not None and os.path.exists(cached_token_path):
             self.logger.info(f"Found cache file in {cached_token_path}, attempting to read the token from it.")
             try:
                 self._auth_session = self.http_session_class()
                 await _set_logged_in_state(username, _get_token_from_cache(cached_token_path, password))
-                return True
+                return self
```

Comment:
The current example.py relies on the output to determine success/failure. Please align the example.py 'login' usage and it's ready :)